src/_pytest/warning_types.py

```diff
@@ -42,6 +42,13 @@ class PytestCollectionWarning(PytestWarning):
     __module__ = "pytest"
 
 
+@final
+class PytestReturnNotNoneWarning(PytestWarning):
```

Comment:
Let's inherit from `PytestDeprecationWarning`, since we want to make this an error in future.

Reply:
>> Looks good to me - if you can confirm that this doesn't trigger when used with `pytest-twisted`, I'll be happy to merge 🥳

I saw there is a test file `pytest_twisted_integration.py`. Does it test `pytest-twisted`? Or do I have to create new tests?

Reply:
I don't know, actually 😝

Without looking, I'd have gone and installed `pytest-twisted`, ran the `inlineCallbacks` and `ensureDeferred` examples to check that they work with the `main` branch and also (without warnings) on this PR.  It looks like that test file does pretty much exactly that though, so since it's passing we're all good 💪